### ISSUE TITLE
'let' inconsistent behavior 

### ISSUE THREAD

**Issue by artelk** [linq2db/linq2db#3181]
The following works fine:
```cs
var q = from t in db.Tbl1
        select new Tbl1
        {
            C1 = t.C1,
            C2 = t.C2
        };

q = from t in q
    select new Tbl1
    {
        C1 = t.C1,
        C2 = t.C2,
        C3 = t.C3
    };

var list = q.ToList();
```
The SQL query executed is
```
SELECT
        [t].[C1],
        [t].[C2]
FROM
        [Tbl1] [t]
```
Which is expected because C3 wasn't initialized in the first query so it uses a default value for the C3.

But the following throws an exception "LinqToDB.Linq.LinqException: Invalid expression <>h__TransparentIdentifier0.t2.C3"
```cs
var q = from t1 in db.Tbl1
        let t2 = new Tbl1
        {
            C1 = t1.C1,
            C2 = t1.C2
        }
        select new Tbl1
        {
            C1 = t2.C1,
            C2 = t2.C2,
            C3 = t2.C3
        };
```
Expected behavior: to work the same as the first query.

**Comment by jods4** [comment]
What would you expect `C3` to be?
To me it should be `null` but that wouldn't be very useful.

**Comment by artelk** [comment]
Yes, it shoud be null and the sql query should only select C1 and C2 because t2 explicitly assigns only these properies.
More accurately: it should/could execute `result.C3=t2.C3` on the client side _after_ the query is executed. 
Imagine the property C3 is defined like this:
```cs
string _c3;
string C3
{
 get => _c3;
 set => _c3 = value ?? "";
}
```
So the `result.C3` should be an empty string. 
But assuming nobody implements any strange logic in the geters\setters of the properties in the data classes Linq2Db could just omit the assignement `result.C3 = t2.C3` because t2.C3 wasn't assigned.

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`
